src/packages/AbstractClient.ts

```diff
@@ -43,25 +44,25 @@ export abstract class AbstractClient extends EventEmitter {
   constructor(options: DeepgramClientOptions) {
     super();
 
-    let key;
+    if (options.accessToken) {
+      this.accessToken = options.accessToken;
+    }
 
     if (typeof options.key === "function") {
       this.factory = options.key;
-      key = this.factory();
+      this.key = this.factory();
     } else {
-      key = options.key;
+      this.key = options.key;
     }
```

Comment:
Assuming DeepgramClientOptions accessToken can be provided as a string or a function that returns a string, we also need to set the factory here if accessToken comes in as a function

HMU if you have Qs